debugutils/aggregator.go

```diff
@@ -0,0 +1,115 @@
+package debugutils
+
+import (
+	"io"
+	"path/filepath"
+
+	"github.com/solo-io/go-utils/errors"
+	"github.com/solo-io/go-utils/installutils/helmchart"
+	"github.com/solo-io/go-utils/tarutils"
+	"github.com/spf13/afero"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	aggregatorName = "aggregator"
+	logsStr = "logs"
+	resourcesStr = "resources"
+
+)
+
+type Aggregator struct {
+	resourceCollector ResourceCollector
+	logCollector      LogCollector
+	storageClient     StorageClient
+	fs                afero.Fs
+
+	// root directory of the aggregated files
+	dir string
+}
+
+func NewAggregator(resourceCollector ResourceCollector, logCollector LogCollector, storageClient StorageClient, fs afero.Fs, dir string) *Aggregator {
+	return &Aggregator{resourceCollector: resourceCollector, logCollector: logCollector, storageClient: storageClient, fs: fs, dir: dir}
+}
+
+func DefaultAggregator() (*Aggregator, error) {
+	fs := afero.NewOsFs()
+	storageClient := NewFileStorageClient(fs)
+	resourceCollector, err := DefaultResourceCollector()
+	if err != nil {
+		return nil, errors.InitializationError(err, aggregatorName)
+	}
+	resourceCollector.storageClient = storageClient
```

Comment:
This is a bit hacky, it means we're using the wrong constructor. The aggregator requires the resource and log collector are implemented with the same underlying storage client (or at least, file system) or it won't work. To me this means:
1) We should never use the Default constructors from the aggregator, 2) We should change the collectors to take a storage client into the storage-related methods, or 3) change the collector interfaces to focus on collecting and split off the saving bits into other interfaces.